lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -6676,376 +6676,362 @@ using namespace Js;
         return newObj;
     }
 
-    struct CompareVarsInfo
-    {
-        ScriptContext* scriptContext;
-        RecyclableObject* compFn;
+    // String Item struct, used for Array.prototype.sort without a compare Function
+    // Spec requires that comparisons are done using a string-comparison
+    // We convert the array in advance into pairs of values and strings
+    // This reduces the total number of conversions needed
+    struct StringItem
+    {
+        Field(Var) Value;
+        Field(JavascriptString*) StringValue;
     };
 
-    int __cdecl compareVars(void* cvInfoV, const void* aRef, const void* bRef)
+    // Comparison method used in Array.prototype.sort when no comparison function was provided
+    bool stringCompare(JavascriptArray::CompareVarsInfo* cvInfo, const void* aRef, const void* bRef)
```

Comment:
Looks like `cvInfo` is not used, can it be dropped?

Reply:
I think my comparisons functions have to have a consistent header - this was my first time using function pointers and I understood that they required them.

`cvInfo->CompareType` is set as a pointer to the relevant comparison method:
- stringCompare (array.sort() with no user compare method)
- compareVarsCrossContext (array.sort with a user compare method from a different scriptContext)
- compareVars (array.sort with a user compare method from the same scriptContext)
- TypedArrayCompareElementsHelper (typedarray.sort)

The sorting algorithms de-reference and call the method from the pointer - all of them other than stringCompare use the cvInfo parameter.